core/src/main/scala/kafka/consumer/BaseConsumer.scala

```diff
@@ -0,0 +1,72 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.consumer
+
+import java.util.Properties
+
+/**
+ * A base consumer used to abstract both old and new consumer
+ * this class should be removed (along with BaseProducer) be removed
+ * once we deprecate old consumer
+ */
+trait BaseConsumer {
+  def receive(): BaseConsumerRecord
+  def close()
+}
+
+case class BaseConsumerRecord(topic: String, partition: Int, offset: Long, key: Array[Byte], value: Array[Byte])
+
+class NewShinyConsumer(topic: String, consumerProps: Properties) extends BaseConsumer {
+  import org.apache.kafka.clients.consumer.KafkaConsumer
+
+  val consumer = new KafkaConsumer[Array[Byte], Array[Byte]](consumerProps)
+  consumer.subscribe(topic)
+  var recordIter = consumer.poll(0).iterator()
+
+  override def receive(): BaseConsumerRecord = {
+    while (!recordIter.hasNext)
+      recordIter = consumer.poll(0).iterator()
```

Comment:
We could change it to be that way around. I'd say it's probably was 6 of one or 1/2 dozen of the other (the loop will block here or, if we specify the timeout, it'll block inside the kafka consumer)? Happy to change if you feel strongly. I actually took this straight from GZ's patch.